### ISSUE TITLE
proble with stack overflow in recursive children

### ISSUE THREAD

**Issue by weepy** [AmpersandJS/ampersand-state#181]
Hi I'm trying to model a (multiply) nested object using `children`

I (obviously) get a stack overflow with :

```
var M = State.extend({    
    children: {
        inner: function(x) {
            // x
            if( !isEmpty(x) ) {
                return new M()
            }
            return new End() // end
        }
    }
});

var m = new M({ inner: { stuff: true } });
```

Any ideas how to do this ?

**Comment by kamilogorek** [comment]
Could you post requirebin gist with your code?

I'm not sure whether you checked, but `inner` function will get empty object here, as `children` object expect `key: value` pair, where `value` is a `state`/`model` constructor.

**Comment by weepy** [comment]
You're right that's exactly what happens, so it means I can't have polymorphic children. 

My pull request simply does ``` this._initChildren(attrs, options);``` in the constructor so that I can override ```_initChildren``` in my derived class.
